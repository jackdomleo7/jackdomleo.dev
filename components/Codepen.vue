<template>
  <p
    class="codepen"
    data-theme-id="39164"
    data-default-tab="result"
    :data-user="user"
    :data-slug-hash="slugHash"
    :data-preview="clickToLoad.toString()"
    :data-pen-title="penTitle"
  >
    <span>See the Pen <a :href="`https://codepen.io/${user}/pen/${slugHash}`">
      {{ penTitle }}</a> by <a :href="`https://codepen.io/${user}`">@{{ user }}</a>
      on <a href="https://codepen.io">CodePen</a>.</span>
  </p>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'Codepen',
  props: {
    clickToLoad: {
      type: Boolean,
      default: false
    },
    user: {
      type: String,
      default: 'jackdomleo7'
    },
    slugHash: {
      type: String,
      required: true
    },
    penTitle: {
      type: String,
      required: true
    }
  }
})
</script>

<style lang="scss">
.cp_embed_iframe,
.codepen {
  height: 40rem;
  max-height: 80vh;
  min-height: 20rem;
  width: 100%;
  display: grid;
  place-items: center;
  border: 1px solid var(--colour-text-secondary);
}
</style>