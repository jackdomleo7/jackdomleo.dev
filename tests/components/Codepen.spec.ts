import { mount, shallowMount } from '@vue/test-utils'
import { configureAxe, toHaveNoViolations } from 'jest-axe'
import Codepen from '@/components/Codepen.vue'

expect.extend(toHaveNoViolations)

const axe = configureAxe({
  rules: {
    region: { enabled: false } // Don't need to check because components are loaded in isolation
  }
})

describe('Codepen', () => {
  /**
   * Vue instance
   */

  it('is a Vue instance', () => {
    const wrapper = mount(Codepen, {
      propsData: {
        slugHash: 'wvoYjNP',
        penTitle: 'Custom car colour preview'
      }
    })

    expect(wrapper.vm).toBeTruthy()
  })

  /**
   * Snapshots
   */

  it('matches snapshot when mounted', () => {
    const wrapper = mount(Codepen, {
      propsData: {
        slugHash: 'wvoYjNP',
        penTitle: 'Custom car colour preview'
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  it('matches snapshot when shallow mounted', () => {
    const wrapper = shallowMount(Codepen, {
      propsData: {
        slugHash: 'wvoYjNP',
        penTitle: 'Custom car colour preview'
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  /**
   * A11y
   */

  it('to have no obvious accessibility violations', async () => {
    const wrapper = mount(Codepen, {
      propsData: {
        slugHash: 'wvoYjNP',
        penTitle: 'Custom car colour preview'
      }
    })

    await expect(axe(wrapper.element)).resolves.toHaveNoViolations()
  })

  /**
   * Props
   */

  it('sets to preview mode', () => {
    const wrapper = shallowMount(Codepen, {
      propsData: {
        slugHash: 'wvoYjNP',
        penTitle: 'Custom car colour preview',
        clickToLoad: true
      }
    })

    expect(wrapper.find('p.codepen').attributes('data-preview')).toBe('true')
  })
})
