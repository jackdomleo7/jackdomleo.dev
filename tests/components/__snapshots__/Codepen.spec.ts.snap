// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Codepen matches snapshot when mounted 1`] = `
<p
  class="codepen"
  data-default-tab="result"
  data-pen-title="Custom car colour preview"
  data-preview="false"
  data-slug-hash="wvoYjNP"
  data-theme-id="39164"
  data-user="jackdomleo7"
>
  <span>
    See the Pen 
    <a
      href="https://codepen.io/jackdomleo7/pen/wvoYjNP"
    >
      
    Custom car colour preview
    </a>
     by 
    <a
      href="https://codepen.io/jackdomleo7"
    >
      @jackdomleo7
    </a>
    
    on 
    <a
      href="https://codepen.io"
    >
      CodePen
    </a>
    .
  </span>
</p>
`;

exports[`Codepen matches snapshot when shallow mounted 1`] = `
<p
  class="codepen"
  data-default-tab="result"
  data-pen-title="Custom car colour preview"
  data-preview="false"
  data-slug-hash="wvoYjNP"
  data-theme-id="39164"
  data-user="jackdomleo7"
>
  <span>
    See the Pen 
    <a
      href="https://codepen.io/jackdomleo7/pen/wvoYjNP"
    >
      
    Custom car colour preview
    </a>
     by 
    <a
      href="https://codepen.io/jackdomleo7"
    >
      @jackdomleo7
    </a>
    
    on 
    <a
      href="https://codepen.io"
    >
      CodePen
    </a>
    .
  </span>
</p>
`;
